core/Enclave/enclave_hsm.cpp

```diff
@@ -294,43 +327,92 @@ sgx_status_t enclave_create_rsa_key(uint8_t *cmk_blob,
         printf("ecall create_rsa_key cmk_keyblob_size:%lu < key_blob_size:%d.\n", cmk_blob_size, real_keyblob_size);
         return SGX_ERROR_INVALID_PARAMETER;
     }
-        rsa_key_data_t *keyblob = (rsa_key_data_t *)cmk_blob;
+    if (keyspec == NULL || padding_mode == NULL)
+        return SGX_ERROR_INVALID_PARAMETER;
+
+    RSA     *rsa            = NULL;
+    BIGNUM  *bne            = NULL;
+    BIO     *public_key     = NULL;
+    BIO     *private_key    = NULL;
+    char    *p_public_key   = NULL;
+    char    *p_private_key  = NULL;
+    string payload;
```

Comment:
Can use memory directly?  string does not make sense.

Reply:
string here is used for a strcat for rsa keypair, I can modify it if necessary